File: kmp_app/composeApp/src/commonMain/kotlin/org/example/project/screens/QuotesScreen.kt
package org.example.project.screens

import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.FormatQuote
import androidx.compose.material.icons.filled.Movie
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinproject.composeapp.generated.resources.Res
import kotlinproject.composeapp.generated.resources.cinema
import org.example.project.models.Quote
import org.example.project.services.QuotesService
import org.example.project.widgets.QuoteCard
import org.jetbrains.compose.resources.painterResource

@Composable
fun QuotesScreen(
    quotesService: QuotesService = QuotesService(),
    onQuoteTap: ((Quote) -> Unit)? = null,
    refreshTrigger: Int = 0
) {
    var quotes by remember { mutableStateOf<List<Quote>>(emptyList()) }
    var isLoading by remember { mutableStateOf(true) }
    var error by remember { mutableStateOf<String?>(null) }

    LaunchedEffect(refreshTrigger) {
        try {
            isLoading = true
            error = null
            quotes = quotesService.getQuotes()
            isLoading = false
        } catch (e: Exception) {
            error = e.message
            isLoading = false
        }
    }

    Column {
        // Header image section with fallback
        Image(
            painter = painterResource(Res.drawable.cinema),
            contentDescription = "Kino Hintergrundbild",
            modifier = Modifier.height(200.dp),
            contentScale = ContentScale.Crop,
        )

        // Content section
        when {
            isLoading -> {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    CircularProgressIndicator()
                }
            }
            error != null -> {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    Text(text = "Error: $error")
                }
            }
            quotes.isEmpty() -> {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    Column(
                        horizontalAlignment = Alignment.CenterHorizontally
                    ) {
                        Icon(
                            imageVector = Icons.Default.FormatQuote,
                            contentDescription = null,
                            modifier = Modifier.size(50.dp)
                        )
                        Spacer(modifier = Modifier.height(16.dp))
                        Text(
                            text = "Keine Zitate vorhanden",
                            fontSize = 24.sp
                        )
                        Spacer(modifier = Modifier.height(8.dp))
                        Text(text = "Fügen Sie Ihre Lieblingszitate aus Filmen hinzu")
                    }
                }
            }
            else -> {
                LazyColumn(
                    modifier = Modifier.fillMaxSize(),
                    contentPadding = PaddingValues(16.dp),
                    verticalArrangement = Arrangement.spacedBy(8.dp)
                ) {
                    items(quotes) { quote ->
                        QuoteCard(
                            quote = quote,
                            onTap = { onQuoteTap?.invoke(quote) }
                        )
                    }
                }
            }
        }
    }
}

File: kmp_app/quotes_screen.dart
import 'package:flutter/material.dart';

import 'package:flutter_app/models/quote.dart';
import 'package:flutter_app/screens/quote_detail_screen.dart';
import 'package:flutter_app/services/quotes_service.dart';
import 'package:flutter_app/widgets/quote_card.dart';

class QuotesScreen extends StatefulWidget {
  const QuotesScreen({super.key});

  @override
  State<QuotesScreen> createState() => QuotesScreenState();
}

class QuotesScreenState extends State<QuotesScreen> {
  final QuotesService _quotesService = QuotesService();
  late Future<List<Quote>> _quotesFuture;

  @override
  void initState() {
    super.initState();
    _loadQuotes();
  }

  void _loadQuotes() {
    _quotesFuture = _quotesService.getQuotes();
  }

  void refreshQuotes() {
    setState(() {
      _loadQuotes();
    });
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Image.asset(
          'assets/images/cinema.png',
          height: 300,
          width: double.infinity,
          fit: BoxFit.fill,
          semanticLabel: 'Kino-Illustration mit Menschen, die Filme schauen',
          errorBuilder: (context, error, stackTrace) {
            // Fallback if image is not found
            return Container(
              height: 200,
              decoration: const BoxDecoration(color: Color(0xFFF3E5F5)),
              child: const Center(
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Icon(Icons.movie, size: 64, color: Colors.black54),
                    SizedBox(height: 8),
                    Text(
                      'Kino & Filmzitate',
                      style: TextStyle(
                        fontSize: 18,
                        fontWeight: FontWeight.bold,
                        color: Colors.black87,
                      ),
                    ),
                  ],
                ),
              ),
            );
          },
        ),
        FutureBuilder<List<Quote>>(
          future: _quotesFuture,
          builder: (context, snapshot) {
            if (snapshot.connectionState == ConnectionState.waiting) {
              return const Center(child: CircularProgressIndicator());
            } else if (snapshot.hasError) {
              return Center(child: Text('Error: ${snapshot.error}'));
            } else if (!snapshot.hasData || snapshot.data!.isEmpty) {
              return const Center(
                child: Column(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: [
                    Icon(Icons.format_quote, size: 50),
                    SizedBox(height: 16),
                    Text(
                      'Keine Zitate vorhanden',
                      style: TextStyle(fontSize: 24),
                    ),
                    SizedBox(height: 8),
                    Text('Fügen Sie Ihre Lieblingszitate aus Filmen hinzu'),
                  ],
                ),
              );
            }
            final quotes = snapshot.data!;
            return Expanded(
              child: ListView.builder(
                padding: const EdgeInsets.all(16),
                itemCount: quotes.length,
                itemBuilder: (context, index) {
                  final quote = quotes[index];
                  return QuoteCard(
                    quote: quote,
                    onTap: () async {
                      final result = await Navigator.of(context).push(
                        MaterialPageRoute(
                          builder: (context) => QuoteDetailScreen(quote: quote),
                        ),
                      );
                      // Refresh the quotes list if something was changed or deleted
                      if (result != null) {
                        refreshQuotes();
                      }
                    },
                  );
                },
              ),
            );
          },
        ),
      ],
    );
  }
}
